### PR TITLE
add Vagrantfile and Dockerfile, prepare to publish packagecloud.io

### PR DESCRIPTION
**Pull request by Slach**
replace outbrain to github in golang url
add packageloud-ruby into Vagrant
little bit code clear
tested packagecloud repo https://packagecloud.io/Slach/orchestrator-agent

Signed-off-by: Slach <bloodjazman@gmail.com>

### CODE REVIEW

File: script/packagecloud-push.rb [hunk 1]
```diff
@@ -0,0 +1,117 @@
+# Pushes all deb and rpm files from ./repos to PackageCloud.
+
+packagecloud_user = ENV["PACKAGECLOUD_USER"] || "github"
+packagecloud_repo = ENV["PACKAGECLOUD_REPO"] || "orchestrator"
```
Comment (shlomi-noach) [inline script/packagecloud-push.rb]: Wrong repository.

File: script/packagecloud-prune.sh [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+PACKAGECLOUD_USER="${PACKAGECLOUD_USER:-github}"
+PACKAGECLOUD_REPO="${PACKAGECLOUD_REPO:-orchestrator}"
```
Comment (shlomi-noach) [inline script/packagecloud-prune.sh]: Wrong repository.

Comment (shlomi-noach) [reply script/packagecloud-prune.sh]: I'd suggest not using `packagecloud` at this time.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    && apk add --update --no-cache bash \
+    && apk add --update --no-cache --virtual .build-deps \
+        rsync \
+        git \
+        go \
+        build-base \
+    && cd /tmp \
+    && { go get -d github.com/github/orchestrator-agent ; : ; }
+#    && cd $GOPATH/src/github.com/github/orchestrator-agent/ \
+#    && go get ./go/... \
```
Comment (shlomi-noach) [inline Dockerfile]: remove commented lines?

File: go/osagent/osagent.go [hunk 4]
```diff
@@ -349,15 +349,15 @@ func LogicalVolumes(volumeName string, filterPattern string) ([]LogicalVolume, e
 		return nil, err
 	}
 
-	logicalVolumes := []LogicalVolume{}
+	var logicalVolumes []LogicalVolume
```
Comment (shlomi-noach) [inline go/osagent/osagent.go]: I prefer it the previous way

File: go/osagent/osagent.go [hunk 5]
```diff
@@ -349,15 +349,15 @@ func LogicalVolumes(volumeName string, filterPattern string) ([]LogicalVolume, e
...
 	for _, lineTokens := range tokens {
 		logicalVolume := LogicalVolume{
 			Name:      lineTokens[1],
 			GroupName: lineTokens[2],
 			Path:      lineTokens[3],
 		}
 		logicalVolume.SnapshotPercent, err = strconv.ParseFloat(lineTokens[4], 32)
-		logicalVolume.IsSnapshot = (err == nil)
+		logicalVolume.IsSnapshot = err == nil
```
Comment (shlomi-noach) [inline go/osagent/osagent.go]: I prefer it the previous way.
I assume some linter said it's not idiomatic or something? To me it's clearer the original way.

**Review by shlomi-noach — CHANGES_REQUESTED** [review]
Remove any packagecloud files for now; some other minor comments.

Thank you for working on this and for the cleanup!

File: script/packagecloud-prune.sh (hunk 2 above)
Comment (Slach) [reply script/packagecloud-prune.sh]: i use packagecloud for my own repo for testing orchestrator-agent on my staging stang
and use packagecloud for install orchestrator

would i rename PACKAGECLOUD_REPO to default orchestrator-agent?

**Comment by shlomi-noach** [comment]
Please remove packagecloud scripts.

**Comment by Slach** [comment]
@shlomi-noach could you merge this PR ? i remove packagecloud scipts

**Review by shlomi-noach — APPROVED** [review]

**Comment by shlomi-noach** [comment]
@Slach to be able to test this I'd need you to make the pr from a branch other than `master`. Can you please `git checkout master -b slach-orchestrator-agent-updates` or similar, and push the new branch?

**Comment by Slach** [comment]
@shlomi-noach 
ok. i create new branch
https://github.com/Slach/orchestrator-agent/tree/add_docker_and_vagrant

could you edit this pull request?

**Comment by shlomi-noach** [comment]
I unfortunately cannot edit the _source_ of the PR, only the _target_.

**Comment by shlomi-noach** [comment]
superseded by https://github.com/github/orchestrator-agent/pull/19 ; https://github.com/github/orchestrator-agent/pull/19 is merged.

**Pull request closed without merging**